src/PlusDataCollection/OptiTrack/vtkPlusOptiTrack.cxx

```diff
@@ -313,21 +319,25 @@ PlusStatus vtkPlusOptiTrack::InternalCallback(sFrameOfMocapData* data)
 {
   LOG_TRACE("vtkPlusOptiTrack::InternalCallback");
 
-  this->Internal->MatchTrackedTools();
-
   const double unfilteredTimestamp = vtkPlusAccurateTimer::GetSystemTime();
 
-  sDataDescriptions* dataDescriptions;
-  this->Internal->NNClient->GetDataDescriptions(&dataDescriptions);
-
-  int numberOfRigidBodies = data->nRigidBodies;
-  sRigidBodyData* rigidBodies = data->RigidBodies;
+  if (this->Internal->LastMotiveDataDescriptionsUpdateTimestamp < 0)
+  {
+    // do an initial match of tracked tools
+    this->Internal->MatchTrackedTools();
+    this->Internal->LastMotiveDataDescriptionsUpdateTimestamp = unfilteredTimestamp;
```

Comment:
This line should be moved into MatchTrackedTools. Maybe MatchTrackedTools would need to be renamed to UpdateMotiveDataDescriptions.